### PR TITLE
Fix interactivity overloads and documentation

### PR DESCRIPTION
**Pull request by IDoEverything**
# Summary
Fixes button interactivity overloads and documentation,

# Details
Fixes an ambiguous call error on `WaitForButtonAsync`, adds `SendPaginatedMessageAsync` with buttons on `DiscordChannel`, and fixes/adds some xml documentation.

### CODE REVIEW

File: DSharpPlus.Interactivity/InteractivityExtension.cs [hunk 1]
```diff
@@ -535,6 +545,16 @@ public async Task<ReadOnlyCollection<Reaction>> CollectReactionsAsync(DiscordMes
...
+        /// <summary>
+        /// Sends a paginated message with buttons.
+        /// </summary>
+        /// <param name="channel">The channel to send it on.</param>
+        /// <param name="user">User to give control.</param>
+        /// <param name="pages">The pages.</param>
+        /// <param name="buttons">Pagination buttons (leave null to default to ones on configuration.</param>
+        /// <param name="behaviour">Pagination behaviour.</param>
+        /// <param name="deletion">Deletion behaviour</param>
+        /// <param name="token">Cancellation token.</param>
```
Comment (VelvetThePanda) [inline DSharpPlus.Interactivity/InteractivityExtension.cs:557]: Ditto

**Review by VelvetThePanda — CHANGES_REQUESTED** [review]

File: DSharpPlus.Interactivity/Extensions/ChannelExtensions.cs [hunk 2]
```diff
@@ -80,14 +82,28 @@ public static Task<InteractivityResult<TypingStartEventArgs>> WaitForUserTypingA
...
+        /// <summary>
+        /// Sends a new paginated message with buttons.
+        /// </summary>
+        /// <param name="channel">Target channel.</param>
+        /// <param name="user">The user that'll be able to control the pages.</param>
+        /// <param name="pages">A collection of <see cref="Page"/> to display.</param>
+        /// <param name="buttons">Pagination buttons (leave null to default to ones on configuration.</param>
+        /// <param name="behaviour">Pagination behaviour.</param>
+        /// <param name="deletion">Deletion behaviour</param>
+        /// <param name="token">Cancellation token.</param>
```
[the remark this answers is not in the document]
Comment (IDoEverything) [reply DSharpPlus.Interactivity/Extensions/ChannelExtensions.cs:102]: Which one? The one on the `InteractivityExtension` also did not have documentation.

[the remark this answers is not in the document]
Comment (VelvetThePanda) [reply DSharpPlus.Interactivity/Extensions/ChannelExtensions.cs:102]: It does https://github.com/DSharpPlus/DSharpPlus/blob/master/DSharpPlus.Interactivity/InteractivityExtension.cs#L131-L133

File: DSharpPlus.Interactivity/Extensions/ChannelExtensions.cs [hunk 3]
```diff
@@ -80,14 +82,28 @@ public static Task<InteractivityResult<TypingStartEventArgs>> WaitForUserTypingA
...
+        /// <param name="channel">Target channel.</param>
+        /// <param name="user">The user that'll be able to control the pages.</param>
+        /// <param name="pages">A collection of <see cref="Page"/> to display.</param>
+        /// <param name="buttons">Pagination buttons (leave null to default to ones on configuration).</param>
+        /// <param name="behaviour">Pagination behaviour.</param>
+        /// <param name="deletion">Deletion behaviour</param>
+        /// <param name="token">A custom cancellation token that can be cancelled at any point.</param>
+        /// <exception cref="InvalidOperationException">Thrown if interactivity is not enabled for the client associated with the channel.</exception>
+        public static Task SendPaginatedMessageAsync(this DiscordChannel channel, DiscordUser user, IEnumerable<Page> pages, PaginationButtons buttons = null, PaginationBehaviour? behaviour = default, ButtonPaginationBehavior? deletion = default, CancellationToken? token = default)
+            => GetInteractivity(channel).SendPaginatedMessageAsync(channel, user, pages,buttons, behaviour, deletion, token);
```
Comment (Neuheit) [inline DSharpPlus.Interactivity/Extensions/ChannelExtensions.cs:105]:
```suggestion
            => GetInteractivity(channel).SendPaginatedMessageAsync(channel, user, pages, buttons, behaviour, deletion, token);
```

**Review by Neuheit — APPROVED** [review]

**Pull request merged**